### ISSUE TITLE
Extending Open-wc

### ISSUE THREAD

**Issue by mercmobily** [open-wc/open-wc#1758]
Hi guys,

We have put together a ton and a half of well documented patterns and code to make SPAs. We would like to make these available to everybody.

We have a set of libraries, but we would rather see them as "kits". Each "kit" will add files to the example application, and modify (with a script) the main component file (so that new links are added to the menu, and the right elements are added as well as server routes etc.).

It's all "kind of" working. However, we are trying to figure out how to package it all.

The basic "kit" application is just an empty shell with a drawer (based on open-wc). Then we have the authentication kit, which adds a bunch of files, a couple of links to the menu (which implies that after installing the kit a script will actually add bits to designated, marked spots in the app) etc. Please note that I really want users to install open-wc and THEN add kits to it.

We are considering all possible scenarios. The first one is to get people to initialise an open-wc project, and then install specific NPM modules that will -- upon installing themselves -- run post-install scripts to do whatever they need to do.

Before we enter that rabbit hole, is there a better way to do this? Consider that we have probably close to 10 "kits", ranging from full login (server/client) to CRUD editing screens, self-updating list, etc.

We are not asking for guidance on how to do this. We are just checking that we won't waste hours making it all work, only to discover that you already provided something that allowed what we are doing in open-wc.

Thank you!

**Comment by mercmobily** [comment]
Sorry if I asked a difficult question! I have a working solution using a monorepo, a npm package with the mono repo, and running `npm install` using a subdirectory of the mono repo. It seems to be working well! However, I am still very open to alternatives that are more open-wc-ish.

**Comment by daKmoR** [comment]
You could extend the [open-wc generator](https://github.com/open-wc/open-wc/tree/master/packages/create) if you are wanna be adventurous 😬
I mean it definitely works and we have some internal variations that output different options/results... however there is little to no documentation so the code would need to be your guide 😅

its not much code as it's quite straight forward with no extra features... still it's not a "consumer product"... or at least we never made it one as interest seemed to be low and alternatives like https://plopjs.com/ are available.

**Comment by mercmobily** [comment]
I didn't know about plop! I will see if I get away with my 10-liners simple solution using npm. Thanks for your help!

**Comment by mercmobily** [comment]
I know this is a closed ticket, but I ended up spending a few hours putting this together for my own use:

https://www.npmjs.com/package/scaffoldizer

I ended up with something very neat. I need to document it, but it won't take long. I am about to write a huge scaffolding system to enrich open-wc apps.

I was also thinking on converting open-wc's init script to this. Let me know if you are at all interested. If it happened, I would make sure scaffoldizer is fully documented as literate code.

Thanks
